File: languages/js/src/entity/identity/did_method.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";

/**
 * Represents the type of method did.
 */
export enum DidMethod {
  /**
   * Represents the polygon id method did.
   */
  PolygonID,
  /**
   * Represents the polygon id test method did.
   */
  PolygonIDTest,
}

export namespace DidMethod {
  export function toProto(method: DidMethod): identityEntitiesProto.DidMethod {
    switch (method) {
      case DidMethod.PolygonID:
        return identityEntitiesProto.DidMethod.POLYGON_ID
      case DidMethod.PolygonIDTest:
        return identityEntitiesProto.DidMethod.POLYGON_ID_TEST
    }
  }

  export function fromProto(
    method: identityEntitiesProto.DidMethod | undefined
  ): DidMethod {
    switch (method) {
      case identityEntitiesProto.DidMethod.POLYGON_ID:
        return DidMethod.PolygonID;
      case identityEntitiesProto.DidMethod.POLYGON_ID_TEST:
        return DidMethod.PolygonIDTest;
      default:
        return DidMethod.PolygonID;
    }
  }
}

File: languages/go/entity/identity/did_method.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"


// DidMethod represents the type of method did.
type DidMethod int32

const (
	// PolygonID represents the polygon id method did.
	PolygonID DidMethod = iota
	// PolygonIDTest represents the polygon id test method did.
	PolygonIDTest DidMethod = iota
)

var (
	DidMethodEnumFromProto = map[proto.DidMethod]DidMethod{
		proto.DidMethod_POLYGON_ID: PolygonID,
		proto.DidMethod_POLYGON_ID_TEST: PolygonIDTest,
	}

	DidMethodEnumToProto = map[DidMethod]proto.DidMethod{
		PolygonID: proto.DidMethod_POLYGON_ID,
		PolygonIDTest: proto.DidMethod_POLYGON_ID_TEST,
	}
)